### PR TITLE
cmath: fix log, mark positional-only args

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: stdlib/2and3/cmath.pyi [hunk 1]
```diff
@@ -14,29 +14,29 @@ if sys.version_info >= (3, 6):
...
-def polar(z: _C) -> Tuple[float, float]: ...
-def rect(r: float, phi: float) -> complex: ...
-def sin(x: _C) -> complex: ...
-def sinh(x: _C) -> complex: ...
-def sqrt(x: _C) -> complex: ...
-def tan(x: _C) -> complex: ...
-def tanh(x: _C) -> complex: ...
+def isinf(__z: _C) -> bool: ...
+def isnan(__z: _C) -> bool: ...
+def log(__x: _C, __y_obj: Optional[_C] = ...) -> complex: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/cmath.pyi]: `None` is indeed the default for the second argument, but if you pass `None` it always throws

```
In [1339]: cmath.log(1, None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Volumes/EXTERNAL/Dropbox/code/taxonomy/taxonomy/db/constants.py in <module>()
----> 1 cmath.log(1, None)

TypeError: must be real number, not NoneType
```
So I think this should not be marked as `Optional`, and perhaps the documentation/signature for the function should be changedd to remove `None` as the claimed default.

Comment (hauntsaninja) [reply stdlib/2and3/cmath.pyi]: Oh that's tricky, will fix. Didn't realise that default value of None could result in different behaviour from passing None in (https://docs.python.org/3/howto/clinic.html#the-null-default-value)... That might slow down these stubtest changes.

Also I could not get this to return -1 https://github.com/hauntsaninja/cpython/blob/3.7/Objects/complexobject.c#L311

Comment (JelleZijlstra) [reply stdlib/2and3/cmath.pyi]: It looks like it returns `-1j` if there is an error, but callers are supposed to call `PyErr_Occurred()` after a call to check for errors; that's what happens in the implementations of `cmath.log`.

Comment (hauntsaninja) [reply stdlib/2and3/cmath.pyi]: Wonder if it'd be reasonable to change https://github.com/hauntsaninja/cpython/blob/3.7/Modules/cmathmodule.c#L965 to check against Py_None as well.
```
~/dev/cpython 3.7 λ rg '(NULL.*Py_None|Py_None.*NULL)' | wc -l
     130
```

Comment (JelleZijlstra) [reply stdlib/2and3/cmath.pyi]: I'd agree with that change, but I feel like at best such a change would only go into 3.9, so it doesn't really solve the issue in the stub.

**Pull request merged**